### ISSUE TITLE
[FEATURE] - Resolve domains to allowed IPs

### ISSUE THREAD

**Issue by SniperAsh6** [zaneschepke/wgtunnel#72]
**Is your feature request related to a problem? Please describe.**

When using a split tunnel approach I'd like to be able to specify a domain which will be resolved upon tunnel connection to add to the allowed IPs. This will save on effectively doing this manually by having to update a dynamic IP in the allowed IPs list.

**Describe the solution you'd like**
To keep the UI clear a new box added to the peer section, allowing a comma separated list of domains (though my use case is only one) which will be iterated through, looked up and the results added to the allowed IPs list.

I'm less keen on this but an alternative which would likely be simpler though is a lot more specific to my use case is where the domain I'm interested in is the same as the endpoint I'm connecting to, have an option to just include that in the allowed IPs.

**Describe alternatives you've considered**

The common alternative is to use a DNS resolver as a part of the tunnel that'll resolve the domain to an address already in the allowed IPs range but this would cause a conflict with something else I've got so the only way I could make that work is have another resolver just for this which I'd prefer to avoid.

**Comment by zaneschepke** [comment]
Hello! Thank you for using the app! Some comments/questions:
- I recommend your endpoint not be (if possible) a dynamic IP address as there is a known issue with WireGuard where it does not re-resolve the endpoint after the initial connection so you will inevitably have connection issues when the IP changes if you do not restart your tunnel after the change. I'm working on a possible solution for this.  
- Is there a reason you cannot add the IP ranges that your domain could point to the desired peer's allowedIps so even if the IP address does change it will still route to the proper peer?

**Comment by SniperAsh6** [comment]
Thank you for getting back to me so quickly, sorry for taking a while to reply. Through some troubleshooting I've managed to find a way around the conflict that was stopping me doing this nicely through DNS so I've managed to solve the problem now and have no issue closing this if it's not something of wider benefit.

To your points, unfortunately the IP changing is unavoidable so this seemed like less effort than having to look up the IP and add it manually. I wouldn't be expecting the IP to change very often (months ideally) so I think natural reconnections would mostly cover it or worst case doing it manually is less than adding the IP myself each time but either way makes discovering it pretty difficult so it's good to hear you're working on something that'd cover it. As to being able to add the possible range in upfront, that's not so easy either as it's not entirely known or otherwise potentially too wide that it'd catch other stuff up in it too.
